### PR TITLE
Preserve constancy and type annotation when tranforming loop closures

### PR DESCRIPTION
**Pull request by Dominator008**

### CODE REVIEW

File: test/com/google/javascript/jscomp/Es6RewriteBlockScopedDeclarationTest.java [hunk 1]
```diff
@@ -408,6 +408,28 @@ public void testLoopClosure() {
...
+            "var $jscomp$loop$0 = {x: undefined};",
+            "for (;;$jscomp$loop$0 = {x: $jscomp$loop$0.x}) {",
+            "  /** @type {number} */ $jscomp$loop$0.x = 3;",
+            "  var f = function($jscomp$loop$0) {",
+            "    return function() { return $jscomp$loop$0.x}",
+            "  }($jscomp$loop$0);",
+            "}"));
+
+    // Preserve inline type annotation and constancy
+    test("for (;;){ const /** number */ x = 3; var f = function() { return x; } }",
```
Comment (MatrixFrog) [inline test/com/google/javascript/jscomp/Es6RewriteBlockScopedDeclarationTest.java]: Let's add one more test where x is a "var" -- I just want to make sure we won't accidentally try to remove something from letConsts that isn't in it.

Comment (Dominator008) [inline test/com/google/javascript/jscomp/Es6RewriteBlockScopedDeclarationTest.java]: vars don't need transpilation I guess :)

Comment (MatrixFrog) [inline test/com/google/javascript/jscomp/Es6RewriteBlockScopedDeclarationTest.java]: So then it would probably be a testSame()

Comment (Dominator008) [inline test/com/google/javascript/jscomp/Es6RewriteBlockScopedDeclarationTest.java]: Done.

**Comment by MatrixFrog** [comment]
Looks like this fixes the handful of NPEs we were seeing. Thank you!

**Pull request merged**